### PR TITLE
Fixes Mech Strafing

### LINKED ISSUE #54727: Mech operators on the default keybind prefs cannot use alt in strafe mode to turn
## Round ID:
Tested on local

## Reproduction:
image

Reset your keybinds to default, which puts block movement on alt
Hop in a mech
toggle strafe mode, tooltip tells you to use alt to temporarily disable it
alt locks you in place because of blocked movement

### PR DESCRIPTION
**Pull request by iamthesilly**
## About The Pull Request

Holding alt to turn while strafing in a mech actually works now.

Closes #54727 

## Why It's Good For The Game
Fixes a thing that was broken

## Changelog
:cl:
fix: You can once again use alt to turn while strafing in a mech
/:cl:

### CODE REVIEW

**Comment by ArcaneDefence** [comment]
https://github.com/tgstation/tgstation/issues/54727
herurgh

**Comment by SapphicOverload** [comment]
> #54727 herurgh

Ah, didn't see there was an issue open for it, thank you

File: code/modules/vehicles/mecha/mecha_mob_interaction.dm [hunk 1]
```diff
@@ -158,12 +158,14 @@
...
 	RegisterSignal(M, COMSIG_MOB_CLICKON, PROC_REF(on_mouseclick), TRUE)
 	RegisterSignal(M, COMSIG_MOB_SAY, PROC_REF(display_speech_bubble), TRUE)
+	RegisterSignal(M, COMSIG_MOVABLE_KEYBIND_FACE_DIR, PROC_REF(on_turn), TRUE)
 	. = ..()
 	update_appearance()
 
 /obj/vehicle/sealed/mecha/remove_occupant(mob/M)
 	UnregisterSignal(M, COMSIG_MOB_CLICKON)
 	UnregisterSignal(M, COMSIG_MOB_SAY)
+	UnregisterSignal(M, COMSIG_MOVABLE_KEYBIND_FACE_DIR)
```
Comment (san7890) [inline code/modules/vehicles/mecha/mecha_mob_interaction.dm:168]: this should really be a single proc-call with a list now

```suggestion
	UnregisterSignal(M, list(
		COMSIG_MOB_CLICKON,
		COMSIG_MOB_SAY,
		COMSIG_MOVABLE_KEYBIND_FACE_DIR,
	))
```

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/mecha/mecha_mob_interaction.dm (hunk 1 above)
Comment (SapphicOverload) [reply code/modules/vehicles/mecha/mecha_mob_interaction.dm:168]: Alright, I'll do that then

**Comment by FlufflesTheDog** [comment]
Works a bit weird with quickstep mechs like the Odysseus, makes a different noise as if trying to turn, but keeps facing the same direction.

**Comment by SapphicOverload** [comment]
> Works a bit weird with quickstep mechs like the Odysseus, makes a different noise as if trying to turn, but keeps facing the same direction.

Fixed

**Pull request merged**